hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/producer/kafka/Producers.java

```diff
@@ -1,40 +1,45 @@
 package pl.allegro.tech.hermes.frontend.producer.kafka;
 
-import org.apache.kafka.clients.producer.Producer;
 import org.apache.kafka.common.Metric;
 import org.apache.kafka.common.MetricName;
-import org.apache.kafka.common.Node;
 import pl.allegro.tech.hermes.api.Topic;
 import pl.allegro.tech.hermes.common.metric.MetricsFacade;
 
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
-import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.function.Predicate;
 import java.util.function.ToDoubleFunction;
+import java.util.stream.Collectors;
 
 // exposes kafka producer metrics, see: https://docs.confluent.io/platform/current/kafka/monitoring.html#producer-metrics
 public class Producers {
-    private final Producer<byte[], byte[]> ackLeader;
-    private final Producer<byte[], byte[]> ackAll;
-
-    private final boolean reportNodeMetrics;
-    private final AtomicBoolean nodeMetricsRegistered = new AtomicBoolean(false);
-
-    public Producers(Producer<byte[], byte[]> ackLeader,
-                     Producer<byte[], byte[]> ackAll,
-                     boolean reportNodeMetrics) {
-        this.ackLeader = ackLeader;
-        this.ackAll = ackAll;
-        this.reportNodeMetrics = reportNodeMetrics;
+    private final KafkaProducer<byte[], byte[]> ackLeader;
+    private final KafkaProducer<byte[], byte[]> ackAll;
+
+    private final List<KafkaProducer<byte[], byte[]>> remoteAckLeader;
+    private final List<KafkaProducer<byte[], byte[]>> remoteAckAll;
+
+
+    public Producers(Tuple localProducers,
+                     List<Tuple> remoteProducers) {
+        this.ackLeader = localProducers.ackLeader;
+        this.ackAll = localProducers.ackAll;
+        this.remoteAckLeader = remoteProducers.stream().map(it -> it.ackLeader).collect(Collectors.toList());
+        this.remoteAckAll = remoteProducers.stream().map(it -> it.ackAll).collect(Collectors.toList());
     }
 
-    public Producer<byte[], byte[]> get(Topic topic) {
+    public KafkaProducer<byte[], byte[]> get(Topic topic) {
         return topic.isReplicationConfirmRequired() ? ackAll : ackLeader;
     }
 
+    public List<KafkaProducer<byte[], byte[]>> getRemote(Topic topic) {
+        return topic.isReplicationConfirmRequired() ? remoteAckLeader : remoteAckAll;
+    }
+
+
+    //TODO: should probably be split per DC
     public void registerGauges(MetricsFacade metricsFacade) {
```

Comment:
Since `pl.allegro.tech.hermes.frontend.producer.kafka.KafkaProducer` is now responsible for instrumentation and is aware of its datacenter, wouldn't it be better to move this method there?

Reply:
Good idea, I've moved metrics to `KafkaMessageSender`. `KafkaMessageSenders` is now only a thin wrapper